src/autoscaler/api/policyvalidator/policy_validator.go

```diff
@@ -0,0 +1,369 @@
+						}
+
+						formatString := "recurring_schedule[{{.j}}] and recurring_schedule[{{.i}}] are overlapping"
+						err := newPolicyValidationError(context, formatString, errDetails)
+						result.AddError(err, errDetails)
+					}
+				}
+			}
+		}
+	}
+}
+
+func (pv *PolicyValidator) validateOverlappingInSpecificDateSchedules(policy *models.ScalingPolicy, specficDateScheduleContext *gojsonschema.JsonContext, result *gojsonschema.Result) {
+	length := len(policy.Schedules.SpecificDateSchedules)
+	var dateTimeRangeList []*DateTimeRange
+	for _, specSched := range policy.Schedules.SpecificDateSchedules {
+		dateTimeRangeList = append(dateTimeRangeList, newDateTimeRange(specSched.StartDateTime, specSched.EndDateTime, policy.Schedules.Timezone))
+	}
+
+	for j := 0; j < length; j++ {
+		for i := j + 1; i < length; i++ {
+			if dateTimeRangeList[j].overlaps(dateTimeRangeList[i]) {
+				context := gojsonschema.NewJsonContext(fmt.Sprintf("%d", j), specficDateScheduleContext)
+				errDetails := gojsonschema.ErrorDetails{
+					"i":                i,
+					"j":                j,
+					"start_date_time1": policy.Schedules.SpecificDateSchedules[j].StartDateTime,
+					"end_date_time1":   policy.Schedules.SpecificDateSchedules[j].StartDateTime,
+					"start_date_time2": policy.Schedules.SpecificDateSchedules[i].StartDateTime,
+					"end_date_time2":   policy.Schedules.SpecificDateSchedules[i].StartDateTime,
+				}
+
+				formatString := "specific_date[{{.j}}]:{start_date_time: {{.start_date_time1}}, end_date_time: {{.end_date_time1}}} and specific_date[{{.i}}]:{start_date_time: {{.start_date_time2}}, end_date_time: {{.end_date_time2}}} are overlapping"
+				err := newPolicyValidationError(context, formatString, errDetails)
+				result.AddError(err, errDetails)
+
+			}
+		}
+	}
+
+}
+
+func getErrorWithJSONMessage(resErr []gojsonschema.ResultError) error {
+	errString := "["
+	for index, desc := range resErr {
+		if index == len(resErr)-1 {
+			errString += fmt.Sprintf("{\"context\": \"%s\", \"description\": \"%s\"}", desc.Context().String(), desc.Description())
+		} else {
+			errString += fmt.Sprintf("{\"context\": \"%s\", \"description\": \"%s\"},", desc.Context().String(), desc.Description())
+		}
+	}
+	errString += "]"
+	return fmt.Errorf(errString)
+}
+
+func hasIntersection(a []int, b []int) bool {
+	m := make(map[int]bool)
+	for _, item := range a {
+		m[item] = true
+	}
+	for _, item := range b {
+		if _, ok := m[item]; ok {
+			return true
+		}
+	}
+	return false
+}
+
+func compareTimesGTEQ(firstTime string, secondTime string) bool {
+	ft, _ := time.Parse(TimeLayout, firstTime)
+	st, _ := time.Parse(TimeLayout, secondTime)
+	if ft.Sub(st) >= 0 {
+		return true
+	}
+	return false
+}
+
+func compareDatesGTEQ(firstDate string, secondDate string) bool {
+	if firstDate == "" {
+		firstDate = "0001-01-01"
+	}
+	if secondDate == "" {
+		secondDate = "0001-01-01"
+	}
+	fd, _ := time.Parse(DateLayout, firstDate)
+	sd, _ := time.Parse(DateLayout, secondDate)
```

Comment:
when end-date is not define, we can't use "0001-01-01" as its default value.   The overlapping check will fail

Reply:
fixed.